pkg/adapter/config/encryptionAdapter.go

```diff
@@ -2,29 +2,155 @@ package config
 
 import (
 	"context"
+	"errors"
+	"github.com/cloudogu/cesapp-lib/keys"
 	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+	"github.com/cloudogu/k8s-dogu-operator/controllers/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type PublicKeyConfigEncryptionAdapter struct {
+	secrets   secret
+	registry  etcdRegistry
+	namespace string
 }
 
-func NewPublicKeyConfigEncryptionAdapter() *PublicKeyConfigEncryptionAdapter {
-	return &PublicKeyConfigEncryptionAdapter{}
+func NewPublicKeyConfigEncryptionAdapter(secretClient secret, registry etcdRegistry, namespace string) *PublicKeyConfigEncryptionAdapter {
+	return &PublicKeyConfigEncryptionAdapter{secrets: secretClient, registry: registry, namespace: namespace}
 }
 
 func (p PublicKeyConfigEncryptionAdapter) Encrypt(
 	ctx context.Context,
 	name common.SimpleDoguName,
 	value common.SensitiveDoguConfigValue,
 ) (common.EncryptedDoguConfigValue, error) {
-	//TODO implement me
-	panic("implement me")
+	pubkey, err := resource.GetPublicKey(p.registry, string(name))
+	if err != nil {
+		return "", domainservice.NewNotFoundError(err, "could not get public key")
+	}
+	encryptedValue, err := pubkey.Encrypt(string(value))
+	if err != nil {
+		return "", domainservice.NewInternalError(err, "could not encrypt value")
+	}
+	return common.EncryptedDoguConfigValue(encryptedValue), nil
 }
 
 func (p PublicKeyConfigEncryptionAdapter) EncryptAll(
 	ctx context.Context,
 	entries map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue,
 ) (map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue, error) {
-	//TODO implement me
-	panic("implement me")
+	encryptedEntries := map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue{}
+	var encryptionErrors []error
+	pubkeys, err := loadPubKeysFromMap(p.registry, entries)
+	if err != nil {
+		return encryptedEntries, domainservice.NewNotFoundError(err, "could not get public key(s)")
+	}
+	for configKey, configValue := range entries {
+		doguname := string(configKey.DoguName)
+		pubkey, _ := pubkeys[doguname]
```

Comment:
🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1005: unnecessary assignment to the blank identifier